### PR TITLE
Nf test tune

### PR DESCRIPTION
**Pull request by alessiovignoli**
Changes aported:

1.  Setting the seed.  Seed can be None but if given is a integer and is given in the tune config. That seed will initialize python, numpy and torch seeds in the TuneWrapper class init. Then numbers will be drown randomly (always the same if the seed is set) and those values will be the seeds for the trials inside tune. So each tune trial/experiment will have is own seed set in a reproducible manner because is dependant on the overall user given seed. 
2. Seed determine also weights initialization for the specific model (1 model per trial). Model are initialized always to same parameters if seed is given. 
3. Added debug mode to explroe weights initialization, seeds and raw output predicgtion on validation set for best model. All this files are outputed only if debug_mode is activated. They will be present in the debug dir under the TuneRun results subdir. This files are used for understanding reproduciobility across identical runs of tune. 
4. handle_tune nf-test has been created and dnatofloat on cpu is set as proxy for reproducibility. More work here in the future
5. Now tune configs can specify a run_params key that handles RunConfig auxiliary information like stop criteria necessary for the FIFOscheduler now present in the dnatofloat cpu tune config.

TODO for the future is decide wich flavor of the nf-test tune should be put as guthub action to test reproducibility.

### CODE REVIEW

File: bin/src/learner/raytune_learner.py [hunk 1]
```diff
@@ -179,12 +210,34 @@ def setup(self, config: dict, training: object, validation: object) -> None:
         self.training = DataLoader(training, batch_size=self.batch_size, shuffle=True)  # TODO need to check the reproducibility of this shuffling
         self.validation = DataLoader(validation, batch_size=self.batch_size, shuffle=True)
 
+        # debug section, first create a dedicated directory for each worker inside Ray_results/<tune_model_run_specific_dir> location
```
[the remark this answers is not in the document]
Comment (mathysgrapotte) [reply bin/src/learner/raytune_learner.py:213]: Or a "reproducibility" mode, as I believe the formula "model + initial state + seed + training code + training data" is our "deep learning" container
